platform/ui/src/components/InputFilterText/InputFilterText.tsx

```diff
@@ -0,0 +1,91 @@
+import classNames from 'classnames';
+import debounce from 'lodash.debounce';
+import React, {
+  ReactElement,
+  useCallback,
+  useEffect,
+  useMemo,
+  useRef,
+  useState,
+} from 'react';
+
+import Icon from '../Icon';
+
+type InputFilterTextProps = {
+  className?: string;
+  value?: string;
+  placeholder: string;
+  onDebounceChange?: (val: string) => void;
+  onChange?: (val: string) => void;
+  debounceTime?: number;
+};
+
+/**
+ * A component to use as the input for text to filter by/on. A debounced callback is automatically provided
+ * so that the filtering in-turn will be debounced. There is also a straight onChange callback when the filter value is
+ * required immediately and NOT debounced. The debounce time is also configurable.
+ */
+const InputFilterText = ({
+  className,
+  value = '',
+  placeholder,
+  onDebounceChange,
+  onChange,
+  debounceTime = 200,
+}: InputFilterTextProps): ReactElement => {
+  const [filterValue, setFilterValue] = useState<string>(value);
+
+  const searchInputRef = useRef(null);
+
+  const debouncedOnChange = useMemo(() => {
+    return debounce(onDebounceChange || (() => {}), debounceTime);
+  }, []);
+
+  // This allows for the filter value to be updated via the props.
+  useEffect(() => setFilterValue(value), [value]);
+
+  useEffect(() => {
+    return debouncedOnChange?.cancel();
+  }, []);
+
+  const handleFilterTextChanged = useCallback(value => {
+    setFilterValue(value);
+
+    if (onChange) {
+      onChange(value);
+    }
+
+    if (onDebounceChange) {
+      debouncedOnChange(value);
+    }
+  }, []);
+
+  return (
+    <label className={classNames('relative', className)}>
```

Comment:
why we need a `relative` here?

Reply:
Because of the absolute positioned child below.